freud/data.py

```diff
@@ -0,0 +1,178 @@
+        self._box = freud.box.Box.from_box(box)
+        self._basis_positions = basis_positions
+
+    def to_system(self, num_replicas=1, scale=1, sigma_noise=0, seed=None):
+        """Generate a system from the unit cell.
+
+        The box and the positions are expanded by ``scale``, and then Gaussian
+        noise with standard deviation ``sigma_noise`` is added to the
+        positions.  All points are wrapped back into the box before being
+        returned.
+
+        Args:
+            num_replicas (:class:`tuple` or int):
+                If provided as a single number, the number of replicas in all
+                dimensions. If a tuple, the number of times replicated in each
+                dimension. Must be of the form (nx, ny, 1) for 2D boxes
+                (Default value = 1).
+            scale (float):
+                Factor by which scale the unit cell (Default value = 1).
+            sigma_noise (float):
+                The standard deviation of the normal distribution used to add
+                noise to the positions in the system (Default value = 0).
+            seed (int):
+                If provided, used to seed the random noise generation. Not used
+                unless ``sigma_noise`` > 0 (Default value = None).
+
+        Returns:
+            tuple (:class:`freud.box.Box`, :class:`np.ndarray`):
+                A system-like object (see
+                :class:`~freud.locality.NeighborQuery.from_system`.).
+        """
+        try:
+            nx, ny, nz = num_replicas
+        except TypeError:
+            nx = ny = num_replicas
+            nz = 1 if self.box.is2D else num_replicas
+
+        if self.box.is2D and nz != 1:
+            raise ValueError("The number of replicas in z must be 1 for a "
+                             "2D unit cell.")
+
+        if any([n > 1 for n in (nx, ny, nz)]):
+            pbuff = freud.locality.PeriodicBuffer()
+            pbuff.compute((self.box, self.basis_positions),
+                          buffer=(nx-1, ny-1, nz-1),
+                          images=True)
+            box = pbuff.buffer_box*scale
+            positions = np.concatenate((self.basis_positions,
+                                        pbuff.buffer_points))
+        else:
+            box = self.box*scale
+            positions = self.basis_positions.copy()
+        positions *= scale
+
+        if sigma_noise != 0:
+            if seed is not None:
+                random_state = np.random.get_state()
+                np.random.seed(seed)
+            mean = [0]*3
+            var = sigma_noise*sigma_noise
+            cov = np.diag([var, var, var if self.dimensions == 3 else 0])
+            positions += np.random.multivariate_normal(
+                mean, cov, size=positions.shape[:-1])
+            if seed is not None:
+                np.random.set_state(random_state)
+
+        positions = box.wrap(positions)
+        return box, positions
```

Comment:
I think the return type should be `RawPoints`. We should accept `(box, points)` as inputs to things but I think it's more appropriate to return a `RawPoints` object than a tuple when we intend to give the user a system-like quantity.

Reply:
After discussing, we decided against making this change now. We can revisit at a later point if we'd like to, particularly if we actually create a parent `System` class to encapsulate these objects.